src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -322,6 +329,21 @@ static int calculateTotalNumToCut(final double fdrThreshold, final long unfilter
     }
 
     /**
+     *  Add the genotype:vc entry into the given map, but fail if a key already exists.
+     *
+     * @param genotypesToConsiderForFiltering SortedMap, changed in place
+     * @param genotype key to insert into the sorted map
+     * @param vc value to insert into the sorted map
+     */
+    private static void putDisallowingKeyOverwrite(SortedMap<UniqueIDWrapper<Genotype>, VariantContext> genotypesToConsiderForFiltering, Genotype genotype, VariantContext vc) {
+
+        final VariantContext vcTemp = genotypesToConsiderForFiltering.putIfAbsent(new UniqueIDWrapper<>(genotype), vc);
+        if (vcTemp != null) {
+            throw new GATKException.ShouldNeverReachHereException("This error may disappear on a subsequent run of this tool, so please try running this tool again.  Otherwise, this can only be fixed by a GATK developer.  Attempting to overwrite a key in the genotypes for filtering map.  This means that genotypePArtifactComparator is (still) not generating unique keys.  See https://github.com/broadinstitute/gatk/issues/3291");
```

Comment:
This comment is out of date now, but it's not a bad idea to keep the check in place.